src/hsmm_mvpy/models.py

```diff
@@ -191,101 +204,139 @@ def fit_single(self, n_events=None, magnitudes=None, parameters=None, threshold=
             if magnitudes is None:
                 magnitudes = np.zeros((n_events,self.n_dims), dtype=np.float64)
             initial_m = magnitudes
-        
-        if starting_points > 1:
-            parameters = [initial_p]
-            magnitudes = [initial_m]
-            if method == 'random':
-                for _ in np.arange(starting_points):
-                    proposal_p = self.gen_random_stages(n_events, self.mean_d)
-                    proposal_m = np.zeros((n_events,self.n_dims), dtype=np.float64)#Mags are NOT random but always 0
-                    proposal_p[parameters_to_fix] = initial_p[parameters_to_fix]
-                    proposal_m[magnitudes_to_fix] = initial_m[magnitudes_to_fix]
-                    parameters.append(proposal_p)
-                    magnitudes.append(proposal_m)
-            elif method == 'grid':
-                parameters = self._grid_search(n_events+1, iter_limit=starting_points, method='grid')
-                magnitudes = np.zeros((len(parameters), n_events, self.n_dims), dtype=np.float64)
-            else:
-                raise ValueError('Unknown starting point method requested, use "random" or "grid"')
-            with mp.Pool(processes=self.cpus) as pool:
-                estimates = pool.starmap(self.EM, 
-                    zip(itertools.repeat(n_events), magnitudes, parameters, itertools.repeat(1),\
-                    itertools.repeat(magnitudes_to_fix),itertools.repeat(parameters_to_fix), itertools.repeat(min_iteration)))   
+        if starting_points > 1 or len(np.shape(magnitudes)) == 3 or len(np.shape(parameters)) == 3:
+            filterwarnings('ignore', 'Convergence failed, estimation hitted the maximum ', )#will be the case but for a subset only hence ignore
+            if len(np.shape(initial_m)) == 2:
+                parameters = [initial_p]
+                magnitudes = [initial_m]
+                if method == 'random':
+                    for _ in np.arange(starting_points):
+                        proposal_p = self.gen_random_stages(n_events, self.mean_d)
+                        proposal_p[parameters_to_fix] = initial_p[parameters_to_fix]
+                        #proposal_m = self._gen_mags(n_events, starting_points, method='random')
+                        # proposal_m[magnitudes_to_fix] = initial_m[magnitudes_to_fix]
+                        # magnitudes.append(proposal_m)
+                        parameters.append(proposal_p)
+                    magnitudes = self._gen_mags(n_events, starting_points, method='random')
+                    magnitudes[:,magnitudes_to_fix,:] = np.tile(initial_m[magnitudes_to_fix], (len(magnitudes), 1, 1))
+
+                elif method == 'grid':
```

Comment:
didn't you want to remove grid?

Reply:
Only for the tutorials. The method will at some point include a grid search on the magnitudes but I don't have time to do it for now